cloud/aws/deploy/batch.go

```diff
@@ -15,13 +15,348 @@
+	})
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Docs: https://docs.aws.amazon.com/batch/latest/userguide/job_definition_parameters.html
+type JobDefinitionContainerProperties struct {
+	Image                string                `json:"image"`
+	ResourceRequirements []ResourceRequirement `json:"resourceRequirements"`
+	Command              []string              `json:"command"`
+	JobRoleArn           string                `json:"jobRoleArn"`
+	ExecutionRoleArn     string                `json:"executionRoleArn"`
+	Environment          []EnvironmentVariable `json:"environment"`
+}
+
+func (p *NitricAwsPulumiProvider) Batch(ctx *pulumi.Context, parent pulumi.Resource, name string, config *deploymentspb.Batch, runtime provider.RuntimeProvider) error {
+	opts := []pulumi.ResourceOption{pulumi.Parent(parent)}
+
+	// Tag the image
+	repo, err := ecr.NewRepository(ctx, name, &ecr.RepositoryArgs{
+		ForceDelete: pulumi.BoolPtr(true),
+		Tags:        pulumi.ToStringMap(tags.Tags(p.StackId, name, "batch")),
+	}, opts...)
+	if err != nil {
+		return err
+	}
+
+	wrappedImage, err := image.NewImage(ctx, name, &image.ImageArgs{
+		SourceImage:   config.GetImage().GetUri(),
+		RepositoryUrl: repo.RepositoryUrl,
+		Runtime:       runtime(),
+		RegistryArgs:  p.RegistryArgs,
+	}, pulumi.Parent(parent), pulumi.DependsOn([]pulumi.Resource{repo}), pulumi.Provider(p.DockerProvider))
+	if err != nil {
+		return err
+	}
+
+	p.BatchRoles[name], err = iam.NewRole(ctx, "BatchJobRole", &iam.RoleArgs{
+		AssumeRolePolicy: pulumi.String(`{
+			"Version": "2012-10-17",
+			"Statement": [
+				{
+					"Action": "sts:AssumeRole",
+					"Principal": {
+						"Service": "ecs-tasks.amazonaws.com"
+					},
+					"Effect": "Allow",
+					"Sid": ""
+				}
+			]
+		}`),
+	}, opts...)
+	if err != nil {
+		return err
+	}
+
+	listActions := []string{
+		// TODO: test that all resources still work without these permissions
+		"sns:ListTopics",
+		"sqs:ListQueues",
+		"dynamodb:ListTables",
+		"s3:ListAllMyBuckets",
+		"tag:GetResources",
+		"apigateway:GET",
+	}
+
+	// This is a tag key unique to this instance of the deployed stack.
+	// Any resource with this unique tag will inherently be scoped to this stack.
+	// This is used to scope the permissions of the lambda to only resources created by this stack.
+	// stackScopedNameKey := tags.GetResourceNameKey(a.stackId)
+
+	// Add resource list permissions
+	// Currently the membrane will use list operations
+	tmpJSON, err := json.Marshal(map[string]interface{}{
+		"Version": "2012-10-17",
+		"Statement": []map[string]interface{}{
+			{
+				"Action":   listActions,
+				"Effect":   "Allow",
+				"Resource": "*",
+				// "Condition": map[string]map[string]string{
+				// 	// Only apply this to resources who have a resource name key that matches this stack
+				// 	"Null": {
+				// 		fmt.Sprintf("aws:ResourceTag/%s", stackScopedNameKey): "false",
+				// 	},
+				// },
```

Comment:
```suggestion
```

Reply:
leaving this here as a reminder to resolve this if we can, its mostly a documented example of something that didn't work but illustrates the issue that needs to be resolved